src/dnx.host/Bootstrapper.cs

```diff
@@ -76,6 +76,7 @@ public Task<int> RunAsync(List<string> args, IRuntimeEnvironment env)
                 serviceProvider.Add(typeof(IRuntimeEnvironment), env);
 
                 CallContextServiceLocator.Locator.ServiceProvider = serviceProvider;
+                var mono = RuntimeEnvironmentHelper.IsMono;
```

Comment:
Add the comment and add an initialize method that sets it up once instead of all the lazy crazyness